langchain/chains/priority_chain.py

```diff
@@ -0,0 +1,104 @@
+"""Chain pipeline where the outputs of one step feed directly into next."""
+import asyncio
+from typing import Any, Dict, List, NamedTuple, Optional, Tuple
+
+from pydantic import Extra, root_validator
+
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForChainRun,
+)
+from langchain.chains.base import Chain
+
+
+class RankingItem(NamedTuple):
+    rank: int
+    chain: Chain
+    timeout: float
+
+
+class PriorityChain(Chain):
+    """Chain where multiple chains are executed in a particular order
+    if the preferred chains time out."""
+
+    rankings: List[RankingItem]
```

Comment:
usability nit: can we just use order of list, without needing to explicitly specify ranking order?

and this may just be personal preference but think folks already find it hard to keep track of all the classes in LangChain. so maybe easier/more straightforward to pass in as `Sequence[Tuple[Chain, float]]` instead of creating a Named tuple. or even having two sep params `chains: Sequence[Chain]` and `timeouts: Sequence[float]`.